conan/tools/files/files.py

```diff
@@ -466,6 +467,11 @@ def untargz(filename, destination=".", pattern=None, strip_root=False, extract_f
                         # https://github.com/conan-io/conan/issues/11065
                         member.linkpath = member.linkpath[len(common_folder) + 1:].replace("\\", "/")
                         member.linkname = member.linkpath
+                if using_long_path_prefix:
+                    # https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry
+                    # File I/O functions in the Windows API convert "/" to "\" as part of converting
+                    # the name to an NT-style name, except when using the "\\?\" prefix
+                    member.name = member.name.replace("/", "\\")
```

Comment:
Sounds good.
The main issue is that processing the above code with mixed forward/backslash was problematic and fragile, so code normalizes to ``/`` to be able to split and process the folders more robustly.
So this, done after the processing should be harmless 🤞